src/renderer.rs

```diff
@@ -1,211 +0,0 @@
-use super::images::Images;
-use crate::gamestate::GameState;
-use crate::util::*;
-use crate::{BOARD_LENGTH, TILE_SIZE};
-use sdl2::gfx::primitives::DrawRenderer;
-use sdl2::pixels::Color;
-use sdl2::rect;
-use sdl2::rect::Rect;
-use sdl2::render::Canvas;
-use sdl2::video::Window;
-
-const TRANSPARENCY: f64 = 0.6;
-
-pub struct Renderer<'a> {
-    canvas: Canvas<Window>,
-    images: Images<'a>,
-}
-
-impl<'a> Renderer<'a> {
-    pub fn new(canvas: Canvas<Window>, images: Images<'a>) -> Result<Self, String> {
-        Ok(Self { canvas, images })
-    }
-
-    pub fn draw(&mut self, gamestate: &GameState) -> Result<(), String> {
-        self.canvas.copy(self.images.get_background(), None, None)?;
-
-        self.draw_selected_piece(gamestate.slected_piece_coord())?;
-
-        if gamestate.in_check() {
-            let king_position = gamestate.get_king();
-            self.draw_king_threatened(king_position)?;
-        }
-
-        let mut moves = gamestate.legal_moves();
-
-        if let Some(point) = gamestate.get_selected_tile() {
-            if moves.contains(&point) {
-                moves.remove(&point);
-                self.draw_selected_tile(point)?;
-            }
-        }
-
-        if let Some(Move { src, dst }) = gamestate.last_move {
-            self.draw_yellow_tile(src)?;
-            self.draw_yellow_tile(dst)?;
-        }
-
-        for (y, x) in moves {
-            if gamestate.is_empty((y, x)) {
-                self.draw_empty_move_tile((x, y))?;
-            } else {
-                self.draw_occupied_move_tile((x, y))?;
-            }
-        }
-        self.draw_tiles(gamestate)?;
-
-        self.draw_moving_piece(gamestate.get_moving_piece())?;
-        self.canvas.present();
-        Ok(())
-    }
-
-    fn draw_tiles(&mut self, gamestate: &GameState) -> Result<(), String> {
-        for i in 0..BOARD_LENGTH {
-            for j in 0..BOARD_LENGTH {
-                self.draw_tile((j, i), gamestate.board[i as usize][j as usize])?;
-            }
-        }
-        Ok(())
-    }
-
-    fn draw_selected_piece(&mut self, coord: Option<Point>) -> Result<(), String> {
-        if let Some((y, x)) = coord {
-            let alpha = 0.95 * 255.0;
```

Comment:
on line 82 you use a constant for transparency. why not here?

Reply:
oh cause i thought all transparency should be the same but i checked lichess which i modeled this on and like the piece you selected has a stronger shade of green

Reply:
Then give it a constant value